### ISSUE TITLE
Unable to find parent doc for class member (where class is not exported)

### ISSUE THREAD

**Issue by bspies-work** [typhonjs-node-esdoc/typhonjs-node-esdoc#2]
I am using this fork of ESDoc to do documentation generation, but am running into an issue where it is unable to find the `parentDoc` (line 849 of DocBuilder.js in the transpiled code, inside the `_getOutputFileName` function) of a static function inside a class. The code where the error is thrown:

```
        case 'get':
          // fal
          var parentDoc = this._find({ longname: doc.memberof })[0];  //parentDoc undefined here
          return this._getOutputFileName(parentDoc);
```

With some print line debugging, I can see that is is the `addEntityPrefix` which is generating the error:

```
Document has no parent doc: src/transport/caches/getEntities.js~EntityCache.addEntityPrefix
E:\Source\GIMLI\web\orion\orion-prep\node_modules\esdoc\out\src\Publisher\Builder\SearchIndexBuilder.js:106
            throw _iteratorError;
            ^

TypeError: Cannot read property 'kind' of undefined
    at SearchIndexBuilder._getOutputFileName (E:\Source\GIMLI\web\orion\orion-prep\node_modules\esdoc\out\src\Publisher\Builder\DocBuilder.js:838:18)
    at SearchIndexBuilder._getOutputFileName (E:\Source\GIMLI\web\orion\orion-prep\node_modules\esdoc\out\src\Publisher\Builder\DocBuilder.js:851:23)
```

My code is structured like the following:

```
import Cache, { testOnlyPreloadCacheWithExplicitData } from './cache';

/** ...comments... **/
class EntityCache extends Cache {
       /** ...comments ... */
       static addEntityPrefix(entityName) { ... }
}
```

where the extended class is:

```
export default class Cache { ... }
```

**Comment by typhonrt** [comment]
I'll take a look soon. At this time since it seems like there is some life back in the main ESDoc effort with a conceivable major release in future months I certainly want to get anyone using my fork to try and use the official ESDoc latest. Of course there is no official release yet on NPM. I don't plan on supporting my fork / branch a long time as they are experimental in nature and the author of ESDoc seems to be committed on moving that project forward to a `v1.0.0` release that supports Babylon.

Basically if we find and confirm a bug in my fork it should be tested against and submitted as an issue to the official esdoc repo if it occurs there as well.

**Comment by bspies-work** [comment]
Understood. Just trying to get our docs working for now :-). The crucial logic here is in the else block of the `_orderedFind` method. This:

```
return data.order('name asec').map(function (v) {
          return v;
        });
```

is returning an empty array.

**Comment by typhonrt** [comment]
As things go I still haven't had too much time to reproduce this error. It would help if you can provide a simple and complete test case if possible as a separate repo or by a code clip in a follow up comment. I tried to reproduce it given the information you provided above.

**Comment by bspies-work** [comment]
OK, so I haven't had much luck with creating a simple reproducible case, but I did notice something which might shed some light on the issue. The `getEntities.js` file follows an unusual pattern in that the EntityCache class is not exported but a static method from the class is: `export default EntityCache.getEntities;` (not the `addEntityPrefix` which is creating the error). The `class/src/transport/caches/getEntities.js~EntityCache.html` never shows up in the output when the bug manifests itself. But when I change the EntityCache class to be exported, this file does show up and the bug disappears.

There also seems to be a case where we export and object literal with functions that return instances of a class that we do not export as well. So it seems to be the general case where we have a reference to an unexported member (class) from an exported one.

**Comment by typhonrt** [comment]
Ah, so a tricky bug / "unusual pattern"! ;)  I'm all for trying to find a way to fix it once there is a reproducible case. Quite likely this bug is in the mainline ESDoc repo as well and a PR should be added there too if it can be fixed in my fork. As things go though I don't have the bandwidth to track down the reproducible case, but if one is provided I'm sure I can fix things rather quickly if you don't arrive at the solution first!

**Comment by bspies-work** [comment]
Rather tricky to do. In my attempt to reproduce, I can easily create a class (not exported) that ESDoc will not generate HTML file for, but getting ESDoc to attempt to process a member of that class is the difficult part. I am pretty sure I can do it IF I can get ESDoc to put an entry like the following in its `search_index.js`:

```
  [
    "src/transport/caches/gettypelists.js~typelistcache.cleartypelistcache",
    "class/src/transport/caches/getTypelists.js~TypelistCache.html#static-method-clearTypelistCache",
    "src/transport/caches/getTypelists.js~TypelistCache.clearTypelistCache",
    "method"
  ],
```

where `TypelistCache` is the class that is not exported. But so far not much luck getting that to happen.
